python/unblob/handlers/executable/elf.py

```diff
@@ -262,6 +375,9 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ElfChunk]:
         if header.e_type == lief.ELF.Header.FILE_TYPE.REL.value:
             end_offset = self.get_signed_kernel_module_end_offset(file, end_offset)
 
+        if self.is_upx(file, start_offset, end_offset):
```

Comment:
You have two functions that checks if it's a UPX file:
- `_ELFBase.is_upx`
- `ElfChunk.is_valid_upx`

I think it would make sense to only use one of them. I would keep `is_valid_upx` and move it outside the class. Use it in both handler and extractor. This way if the UPX header is wrong, we catch it as early as possible, rather than carving an invalid UPX and bailing out at extraction time